api/user.go

```diff
@@ -228,3 +279,78 @@ func (ua *UserAPI) ChangePassword() {
 		ua.CustomAbort(http.StatusInternalServerError, "Internal error.")
 	}
 }
+
+// ToggleUserAdminRole handles Post api/users/{}/toggleadmin
+func (ua *UserAPI) ToggleUserAdminRole() {
+	if !ua.IsAdmin {
+		log.Warningf("current user, id: %d does not have admin role, can not update other user's role", ua.currentUserID)
+		ua.RenderError(http.StatusForbidden, "User does not have admin role")
+		return
+	}
+	userQuery := models.User{UserID: ua.userID}
+	if err := dao.ToggleUserAdminRole(userQuery); err != nil {
```

Comment:
If there's the only reference of toggleuser admin role, it would be better to update to dao.TollgleUserAdminRole(${userid},${adminornot})